tests/BenchmarkDotNet.IntegrationTests/InProcessTest.cs

```diff
@@ -56,10 +57,28 @@ public InProcessTest(ITestOutputHelper output) : base(output)
         [Fact]
         public void BenchmarkActionValueTaskOfTSupported() => TestInvoke(x => x.InvokeOnceValueTaskOfT(), UnrollFactor, DecimalResult);
 
+        [Fact]
+        public void BenchmarkDifferentPlatformReturnsValidationError()
+        {
+            var otherPlatform = IntPtr.Size == 8
+                ? Platform.X86
+                : Platform.X64;
+
+            var otherPlatformConfig = new ManualConfig()
+                .With(Job.Dry.With(InProcessToolchain.Instance).WithInvocationCount(UnrollFactor).WithUnrollFactor(UnrollFactor).With(otherPlatform))
```

Comment:
nit: `Job.Dry` already defines invocation count and unroll factor (1)

```suggestion
                .With(Job.Dry.With(InProcessToolchain.Instance).With(otherPlatform))
```